build.go

```diff
@@ -313,9 +314,18 @@ func Build(builder Builder, options ...Option) {
 		}
 	}
 
+	var disableBOMLabel bool
+	if val, ok := os.LookupEnv("BP_DISABLE_BOM_LABEL"); ok {
```

Comment:
@dmikusa-pivotal I think the usage of this env var might be buildpacks/platform specific. I would prefer it if we didn't introduce it in libcnb and just relied on the buildpack to toggle this logic. This might be better placed in libpak for eg. LMK what you think.

Reply:
I'm not opposed to having a different trigger, but I feel pretty strongly that we should allow for the disabling of the label-based BOM entries. It's known to cause problems.

I went with an env variable because there is some precedent with `BP_DEBUG` and `BP_LOG_LEVEL` which are also in libcnb.

Other options that come to mind:

- A BuildContext property: https://github.com/buildpacks/libcnb/blob/main/build.go#L34-L56
- An Option callback: https://github.com/buildpacks/libcnb/blob/d3e6e18255ecfc1dac6cc473471d9cb2739124c4/config.go#L75

Reply:
@dmikusa-pivotal I think the difference between those log level env vars and the SBOM env var is that the log level env var influences the log lines that the build process outputs which cannot be controlled by the users of libcnb (for eg https://github.com/buildpacks/libcnb/blob/d3e6e18255ecfc1dac6cc473471d9cb2739124c4/build.go#L179) I think in this case, a buildpack using libcnb can just detect this env var and not choose to populate the bom in launch.toml/build.toml

I am concerned because I would like that env var to be set to a value so that labels are turned off by default completely and I imagine paketo might want something else. Given that we are venturing into buildpack author level opinions on configuration that is exposed to the end user(app dev) I think we should avoid introducing this in libcnb.

I might be fine with introducing this in a way that is exposed to the buildpack author and not an end user (i.e. not an env var)

Reply:
✅ , I get where you're coming from.

I removed the env variable, added an option that a buildpack author can set when running `libcnb.Build`, set this to default to false (i.e. don't write the BOM label).

Let me know what you think.